src/components/GeneralLookupField.tsx

```diff
@@ -0,0 +1,210 @@
+import React, { useState, useContext, useCallback } from 'react';
+import { Field, useFormikContext } from 'formik';
+import { ResolutionError } from '@unstoppabledomains/resolution/build/resolutionError';
+
+import { DomainStatus, InlineMessage } from '@components';
+import { Network, IReceiverAddress, ErrorObject } from '@types';
+import { getBaseAssetByNetwork, AssetContext } from '@services/Store';
+import { isValidETHAddress, isValidENSName } from '@services/EthService';
+import UnstoppableResolution from '@services/UnstoppableService';
+import { isValidETHRecipientAddress } from '@services/EthService/validators';
+import { useEffectOnce } from '@vendor';
+
+import GeneralLookupDropdown from './GeneralLookupDropdown';
+
+export interface IGeneralLookupFieldComponentProps {
+  error?: string | ErrorObject;
+  network: Network;
+  isResolvingName: boolean;
+  name: string;
+  value: IReceiverAddress;
+  options: any[];
+  placeholder?: string;
+  onBlur?(): void;
+  setIsResolvingDomain(isResolving: boolean): void;
+  handleEthAddress(inputString: string): IReceiverAddress;
+  handleENSName(resolvedAddress: string, inputString: string): IReceiverAddress;
+  onSelect?(option: IReceiverAddress): void;
+  onChange?(input: string): void;
+  onLoad?(setFieldValue: (field: string, value: any, shouldValidate?: boolean) => void): void;
+}
+
+const GeneralLookupField = ({
```

Comment:
Can you add some tests for this field to make sure name resolution and inputs are correctly validated and displaying correct error and loading states?